metals/src/main/scala/scala/meta/internal/metals/config/RunType.scala

```diff
@@ -0,0 +1,20 @@
+package scala.meta.internal.metals.config
+
+object RunType {
+  sealed trait RunType
+  case object Run extends RunType
+  case object TestFile extends RunType
+  case object TestTarget extends RunType
+
+  def fromString(string: String): Option[RunType] = {
+    string match {
+      case "run" => Some(Run)
+      case "testFile" => Some(TestFile)
```

Comment:
we can have just test and if a target is specified and not file it would mean it's supposed to test all test in target?

Reply:
But then where would the target come from? A user would have to include it? This way, there is no need need, a user can either bind a `testFile` or a `testTarget` to whatever they want, without any need to have to provide a target.

Reply:
Hmm.. that is a tough one. Your solution might be best here if we want to provide the functionality to run all tests in the current target.

One improvement could be that if a user doesn't specify file and type is `testFile` or `testTarget` we could use focusedDocument (or focusedTarget which I think exists.)

Reply:
Yea it's a good point. I'm just not convinced that on the client side if we are using `testFile` or `testTarget`, there shouldn't really be a reason why we can't send along uri since it should be quite easy to. Do you think this might be an issue from VS Code if someone is trying to trigger from like the side panel or something? I don't think it's unreasonable to just expect a uri.

Reply:
We should be able to always send a URI - especially if we start from a command. The only exception would be if we wanted to have a `launch.json` configuration that would just run anything in the current file. This would allow to just use the existing shortcut for debug (F5), but on the other hand we can always add a separate command for that and `launch.json` might be a hack. So maybe leave it out for now and I will see how best to implement it on the vs code side.